ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -0,0 +1,288 @@
+        logger.warning(
+            "Calling unimplemented method set_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    def get_baggage_item(self, key):
+        logger.warning(
+            "Calling unimplemented method get_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    # TODO: Verify calls to deprecated methods `log_event()` and `log()` on
+    # base class work properly (it's probably fine because both methods call
+    # `log_kv()`).
+
+
+class ScopeShim(opentracing.Scope):
+    """A `ScopeShim` wraps the OpenTelemetry functionality related to span
+    activation/deactivation while using OpenTracing `Scope` objects for
+    presentation.
+
+    There are two ways to construct a `ScopeShim` object: using the `span`
+    argument and using the `span_cm` argument. One and only one of `span` or
+    `span_cm` must be specified.
+
+    When calling the initializer while passing a `SpanShim` object in the
+    `span` argument, the `ScopeShim` is initialized as a regular `Scope`
+    object.
+
+    When calling the initializer while passing a context manager *generator* in
+    the `span_cm` argument (as returned by the `use_span()` method of
+    OpenTelemetry `Tracer` objects,  the resulting `ScopeShim` becomes
+    usable as a context manager (using `with` statements).
+
+    It is necessary to have both ways for constructing `ScopeShim`
+    objects because in some cases we need to create the object from a context
+    manager, in which case our only way of retrieving a `Span` object is by
+    calling the `__enter__()` method on the context manager, which makes the
+    span active in the OpenTelemetry tracer; whereas in other cases we need to
+    accept a `SpanShim` object and wrap it in a `ScopeShim`.
+    """
+
+    def __init__(self, manager, span=None, span_cm=None):
+        super().__init__(manager, span)
+        self._span_cm = span_cm
+
+        # If a span context manager is provided, extract the `Span` object from
+        # it, wrap the extracted span and save it as an attribute.
+        if self._span_cm is not None:
+            otel_span = self._span_cm.__enter__()
+            self._span = SpanShim(
+                self._manager.tracer, otel_span.get_context(), otel_span
+            )
+
+    def close(self):
+        if self._span_cm is not None:
+            # We don't have error information to pass to `__exit__()` so we
+            # pass `None` in all arguments. If the OpenTelemetry tracer
+            # implementation requires this information, the `__exit__()` method
+            # on `opentracing.Scope` should be overridden and modified to pass
+            # the relevant values to this `close()` method.
+            self._span_cm.__exit__(None, None, None)
+        else:
+            self._span.unwrap().end()
+        # TODO: Set active span on OpenTelemetry tracer.
```

Comment:
That's what `span_cm.__exit__` does already? See my earlier comment in this review on `active` for the else: `self._span` case.

Reply:
True! I've refined the tests in https://github.com/open-telemetry/opentelemetry-python/pull/211/commits/d6e82bc99e354b19c54a7168a4c81bd3125e5741 to verify this and it looks OK to me.